pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/route/UserCreationRoute.java

```diff
@@ -0,0 +1,168 @@
+        * so it can be referred back to after we create the user
+        */ 
+        final var requestorClientId = auth.getClient();
+        final var operatorId = auth.getOperator();
+        final var domain = auth.getIssuer();
+        final var studyGuid = payload.getStudyGuid();
+
+        // Not sure if this is the proper check, but this is
+        // just to add some sort of safeguard until the filters are
+        // configured properly.
+        //
+        // Could this be replaced with an "isDSM" check?
+        if (false && auth.hasAdminAccessToStudy(studyGuid) == false) {
+            var error = new ApiError(ErrorCodes.INSUFFICIENT_PRIVILEGES,
+                    String.format("User %s does not have sufficient priviliges for %s.",
+                    operatorId, studyGuid));
+            throw ResponseUtil.haltError(HttpStatus.SC_FORBIDDEN, error);
+        }
+
+        final var email = payload.getEmail();
+        final var emailValidator = EmailValidator.getInstance();
+
+        /* 
+        * It's not absolutely necessary to check here- the ParticipantsCreateService will also
+        *  check that the email is correctly formatted (you'd need to update its associated
+        *  switch statement below in the transaction)
+        *
+        * Checking here is in order to save a few cycles & connections to the DB if it's
+        *  immediately clear the client gave us bad data.
+        */
+        if (emailValidator.isValid(email) == false) {
+            var error = new ApiError(ErrorCodes.MALFORMED_EMAIL, "the email address is not in a valid format.");
+            throw ResponseUtil.haltError(HttpStatus.SC_UNPROCESSABLE_ENTITY, error);
+        }
+
+        if (StringUtils.isNotBlank(payload.getCenterId())) {
+            var error = new ApiError(ErrorCodes.NOT_SUPPORTED,
+                    "passing a non-null center id when creating a participant is not supported yet.");
+            throw ResponseUtil.haltError(HttpStatus.SC_UNPROCESSABLE_ENTITY, error);
+        }
+
+        log.info("attempting to create a user for study {}", payload.getStudyGuid());
+
+        return TransactionWrapper.withTxn(handle -> {
+            // If we've made it this far in the request, it's unlikely the Auth0 client is not
+            // going to be recognized but handle things just in case. Save the ID since it'll be
+            // needed below to set the `created_by_client_id` field in the user.
+            final var internalClientId = handle.attach(JdbiClient.class)
+                    .getClientIdByAuth0ClientAndDomain(requestorClientId, domain);
+            if (internalClientId.isEmpty()) {
+                /* Disabled for testing!
+                var error = new ApiError(ErrorCodes.NOT_FOUND,
+                        String.format("Auth0 client '%s' is not authorized for '%s'.",
+                            requestorClientId, domain));
+                throw ResponseUtil.haltError(HttpStatus.SC_UNAUTHORIZED, error);
```

Comment:
Should this be uncommented?

Reply:
See [this comment](https://github.com/broadinstitute/ddp-study-server/pull/1653/files#r902677146)